libgalois/src/PropertyGraph.cpp

```diff
@@ -158,16 +158,29 @@ GetTypeSetIDsFromProperties(
 
   // collect the list of types
   std::vector<int> type_field_indices;
+  using BoolProperty = std::pair<int, std::shared_ptr<arrow::BooleanArray>>;
```

Comment:
For clarity, you should rename and introduce a `struct PropertyColumn { int index_; std::shared_ptr<arrow::BooleanArray> array_; };`
This will add clarity to the usage below. GSG also advises against `std::pair<>` and I hate the GSG but I agree with this clause for most cases.

Reply:
I thought of pair vs. struct. I went with pair in this case because of the for loop iterator:
`for (auto [i, bool_property] : bool_properties)`
That's more concise code than with struct and I thought it was easier to read. But I'm not strongly in favor of it over struct.

Reply:
while the code is concise, it's not clear with `i` is doing and why is this thing called `BoolProperty` when it's a pair.

Reply:
Name would be the same whether it's a struct or a pair. What would you want the name to be? `BoolPropertyColumn`? `Column` is superfluous IMO.

Reply:
I can rename `i` to `field_index`. If you strongly want a struct, I can change it too.

Reply:
@amberhassaan This is urgent. Please confirm/approve.

Reply:
I realize that struct needs a template type for arrow array , so you can reuse the struct for both `arrow::BooleanArray` and `arrow::UInt8Array`.

Reply:
That's too verbose/general IMO but I won't argue about it. Will do.

Reply:
Too busy and pressed to argue? :)
My suggestion is to hide the struct in anonymous namespace so it doesn't escape the .cpp file.

Reply:
Hehe, yes. Small stuff. 

All of this is in anonymous namespace.